### PR TITLE
RN-391: Mobile map legend

### PR DESCRIPTION
**Pull request by edmofro**
### Issue #:
RN-391

### Changes:
Adds a mobile version of the map overlay legend. See screenshots [in the ticket](https://linear.app/bes/issue/RN-391#comment-22fb06e4)

### CODE REVIEW

File: packages/web-frontend/src/containers/mobile/MapOverlayBar/MapOverlayLibrary.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ import Button from '@material-ui/core/Button';
 
 import { setMapOverlays, clearMeasure } from '../../../actions';
 import { selectCurrentMapOverlayCodes } from '../../../selectors';
-import { MAP_OVERLAY_SELECTOR, LEAFLET_Z_INDEX } from '../../../styles';
+import { MAP_OVERLAY_SELECTOR, MOBILE_BACKGROUND_COLOR, LEAFLET_Z_INDEX } from '../../../styles';
```
Comment (edmofro) [inline packages/web-frontend/src/containers/mobile/MapOverlayBar/MapOverlayLibrary.js:16]: There are a few unrelated changes in this PR just tweaking the background colour of most mobile components to match the designs

File: packages/ui-components/src/components/Map/Legend/Legend.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ const LegendFrame = styled.div`
   display: flex;
   width: fit-content;
   padding: 0.6rem;
-  margin: ${p => p.$spaceBetween || '0.6rem'} auto;
+  margin: 0.6rem auto;
```
Comment (IgorNadj) [inline packages/ui-components/src/components/Map/Legend/Legend.js]: Minor: is it better to specify options on presentation components e.g. layout: 'vertical' | 'horizontal' or to allow custom components to be passed in? Happy with either way, just curious. Either way this seems better than spaceBetween

**Review by IgorNadj — APPROVED** [review]
LGTM

File: packages/web-frontend/src/containers/mobile/MapOverlayLegend.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import Button from '@material-ui/core/Button';
+import ExpandLessIcon from '@material-ui/icons/ExpandLess';
+import { Legend } from '@tupaia/ui-components/lib/map';
```
[the remark this answers is not in the document]
Comment (edmofro) [reply packages/web-frontend/src/containers/mobile/MapOverlayLegend.js:14]: @tcaiger actually set it up so that there's code splitting within `@tupaia/ui-components`, so that theoretically the bundle size of any consuming package could be smaller by e.g. only importing the map stuff if it's needed

File: packages/ui-components/src/components/Map/Legend/Legend.js (hunk 2 above)
Comment (edmofro) [reply packages/ui-components/src/components/Map/Legend/Legend.js]: Honestly I'm not sure, I think it's just a flavour that different devs have preferences for. I think the ideal is that you compose it together at the top level, the way that `mui` is set up, but that ends up requiring a lot more work and larger JSX at that top level than just providing "slots" that you can pass custom components through.
